Source/SequenceMethodCall.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq.Expressions;
+using System.Reflection;
+
+using Moq.Proxy;
+
+namespace Moq
+{
+	// this corresponds to a setup created by `mock.SetupSequence`
+	internal class SequenceMethodCall : MethodCall
+	{
+		// contains the responses set up with `CallBase`, `Pass`, `Returns`, and `Throws`
+		// TODO: the use of a queue without keeping track of the number of received calls, and tying each
+		// response to an expected invocation count, means that things won't work quite like they used to!
+		// this is possibly a breaking change right now.
+		private ConcurrentQueue<Tuple<ResponseKind, object>> responses;
+
+		public SequenceMethodCall(Mock mock, Condition condition, Expression originalExpression, MethodInfo method, params Expression[] arguments)
+			: base(mock, condition, originalExpression, method, arguments)
+		{
+			this.responses = new ConcurrentQueue<Tuple<ResponseKind, object>>();
+		}
+
+		public void AddCallBase()
+		{
+			this.responses.Enqueue(Tuple.Create(ResponseKind.CallBase, (object)null));
+		}
+
+		public void AddPass()
+		{
+			this.responses.Enqueue(Tuple.Create(ResponseKind.Pass, (object)null));
+		}
+
+		public void AddReturns(object value)
+		{
+			this.responses.Enqueue(Tuple.Create(ResponseKind.Returns, value));
+		}
+
+		public void AddThrows(Exception exception)
+		{
+			this.responses.Enqueue(Tuple.Create(ResponseKind.Throws, (object)exception));
+		}
+
+		public override void Execute(ICallContext call)
+		{
+			base.Execute(call);
```

Comment:
I've quickly gone over the base implementation of `Execute`. Since the fluent API started by `SetupSequence` doesn't allow setting up any explicit `Callback`s, `Return`s, `Raises`, or `Throws`, the base implementation should be largely inert, except for increasing its own `base.CallCount` and setting `base.Invoked`. But it makes me curious how the new `SetupSequence` implementation interacts with `mock.Verify`? Perhaps add some unit tests about that before and after the change.

Reply:
Done. Added a unit test that ensures `Verify` can correctly check a sequence's invocation count.